aiohttp/web.py

```diff
@@ -332,140 +332,176 @@ def raise_graceful_exit():
-            loop.create_unix_server(
-                handler, path, ssl=ssl_context, backlog=backlog
+            server_creations.append(
+                self.loop.create_server(
+                    handler, host_binding, self.port, ssl=self.ssl_context,
+                    backlog=self.backlog
+                )
             )
-        )
-        uris.append('{}://unix:{}:'.format(scheme, path))
-
-        # Clean up prior socket path if stale and not abstract.
-        # CPython 3.5.3+'s event loop already does this. See
-        # https://github.com/python/asyncio/issues/425
-        if path[0] not in (0, '\x00'):  # pragma: no branch
-            try:
-                if stat.S_ISSOCK(os.stat(path).st_mode):
-                    os.remove(path)
-            except FileNotFoundError:
-                pass
-    for sock in socks:
-        server_creations.append(
-            loop.create_server(
-                handler, sock=sock, ssl=ssl_context, backlog=backlog
+
+        for path in self.paths:
+            # Most loop implementations don't support multiple paths bound in
+            # same server, so create a server for each.
+            server_creations.append(
+                self.loop.create_unix_server(
+                    handler, path, ssl=self.ssl_context, backlog=self.backlog
+                )
+            )
+            uris.append('{}://unix:{}:'.format(self.scheme, path))
+
+            # Clean up prior socket path if stale and not abstract.
+            # CPython 3.5.3+'s event loop already does this. See
+            # https://github.com/python/asyncio/issues/425
+            if path[0] not in (0, '\x00'):  # pragma: no branch
+                try:
+                    if stat.S_ISSOCK(os.stat(path).st_mode):
+                        os.remove(path)
+                except FileNotFoundError:
+                    pass
+
+        for sock in self.socks:
+            server_creations.append(
+                self.loop.create_server(
+                    handler, sock=sock, ssl=self.ssl_context,
+                    backlog=self.backlog
+                )
             )
-        )
 
-        if hasattr(socket, 'AF_UNIX') and sock.family == socket.AF_UNIX:
-            uris.append('{}://unix:{}:'.format(scheme, sock.getsockname()))
-        else:
-            host, port = sock.getsockname()[:2]
-            uris.append(str(base_url.with_host(host).with_port(port)))
-    return server_creations, uris
+            if hasattr(socket, 'AF_UNIX') and sock.family == socket.AF_UNIX:
+                uris.append('{}://unix:{}:'.format(self.scheme,
+                                                   sock.getsockname()))
+            else:
+                host, port = sock.getsockname()
+                uris.append(str(self.base_url.with_host(host).with_port(port)))
+
+        self.uris = uris
+        return asyncio.gather(*server_creations, loop=self.loop)
+
+    @asyncio.coroutine
+    def start(self):
```

Comment:
How could I start specific app in runtime after starting all the apps?

Reply:
I'm not sure what you mean. This is not meant to start selectively the applications. This is meant to group all the target applications and run them together on one connection. I mostly inspired on hapi's Glue for the concept.

Reply:
I mean if Server object manages multiple applications, then why not you can't use it to register/start/stop arbitrary application is runtime? Quite oblivious task for it.

Reply:
I haven't make it with the idea of changing the applications during runtime at all but it sounds like an interesting use case.